### ISSUE TITLE
Fix multiscale labels and magic_name guessing with lables

### ISSUE THREAD

**Issue by tlambert03** [napari/napari#2808]
## 🐛 Bug
After #2807 merges, we should try to a couple issues with labels that were addressed in #2683.  Add these two datasets to `layer_test_data` and run tests to see the fails:

```python
# _tests/utils.py
layer_test_data = [
    ...
    (Labels, np.zeros((10, 10), dtype=bool), 2),
    (Labels, np.random.randint(20, size=(6, 10, 15)), 3),
    (
        Labels,
        [np.random.randint(20, size=s) for s in [(40, 20), (20, 10), (10, 5)]],
        2,
    )
]

```

**Comment by kne42** [comment]
hi @tlambert03, could you please provide more context around what this bug is and specifically where the magic name guessing relates to it? thanks!

**Comment by tlambert03** [comment]
See point 2 from #2638 linked above:

> magic_name works on the assumption that the data object IS the object passed into the constructor. So it won't work with the current MultiScaleData wrapper without some modifications. (there's another latent bug here, since it also wouldn't work multiscale data based on GeneratorTypes, since we call list(data) and create a new object... and it won't work with bool type labels, since we convert those to int arrays... but we don't test those either))

**Comment by tlambert03** [comment]
sorry @kne42  ... you must have been very confused :joy:  I realize now that this has been fixed by re-reverting #2683 in #3560 

https://github.com/napari/napari/pull/3560/files#diff-11f2c63f66f3e9b0271b3c1584a67e2eb682b8638c1c679bcd9d036f977675c5R247-R248
